### PR TITLE
Message factory

### PR DESCRIPTION
**Pull request by tpazderka**
- [x] Any changes relevant to users are recorded in the `CHANGELOG.md`.
- [ ] The documentation has been updated, if necessary.
---

### CODE REVIEW

File: src/oic/oauth2/message.py [hunk 1]
```diff
@@ -1100,7 +1101,42 @@ class ASConfigurationResponse(Message):
 
 
 def factory(msgtype):
+    warnings.warn('`factory` is deprecated. Use `OIDCMessageFactory` instead.', DeprecationWarning)
```
Comment (schlenk) [inline src/oic/oauth2/message.py]: Looks like the wrong warning for the oauth2 module.

Comment (tpazderka) [reply src/oic/oauth2/message.py]: Yep, copy paste error...

**Comment by tpazderka** [comment]
The CI is failing on integration tests due to removed imports. I have already submitted a PR there to fix that so we just need to wait.

**Comment by tpazderka** [comment]
OK, the CI is finally passing. Please review again.

**Pull request merged**